README.md
# flutter_picker_view

A  Flutter picker view like iOS UIPickerView support multiple pickers

## Getting Started

First, add flutter_picker_view as a dependency in your pubspec.yaml file.

## Demo
![image](https://github.com/CodeGeekXu/flutter_picker_view/blob/master/flutter_picker_view.gif)

## Example
```dart
import 'package:flutter_picker_view/flutter_picker_view.dart';
```

```dart
enum PickerShowMode {
    AlertDialog,
    BottomSheet
}
```
### AlertDialog
![image](https://github.com/CodeGeekXu/flutter_picker_view/blob/master/alert_dialog.jpeg)

### BottomSheet
![image](https://github.com/CodeGeekXu/flutter_picker_view/blob/master/bottom_sheet.jpeg)

```dart
void _showPicker() {
    PickerController pickerController = PickerController(count: 3, selectedItems: [5,2,1]);
    
    PickerViewPopup.showMode(
        PickerShowMode.AlertDialog, // AlertDialog or BottomSheet
        controller: pickerController,
        context: context,
        title: Text('AlertDialogPicker',style: TextStyle(fontSize: 14),),
        cancel: Text('cancel', style: TextStyle(color: Colors.grey),),
        onCancel: () {
            Scaffold.of(context).showSnackBar(
                SnackBar(content: Text('AlertDialogPicker.cancel'))
            );
        },
        confirm: Text('confirm', style: TextStyle(color: Colors.blue),),
        onConfirm: (controller) {
            List<int> selectedItems = [];
            selectedItems.add(controller.selectedRowAt(section: 0));
            selectedItems.add(controller.selectedRowAt(section: 1));
            selectedItems.add(controller.selectedRowAt(section: 2));

            Scaffold.of(context).showSnackBar(
                SnackBar(content: Text('AlertDialogPicker.selected:$selectedItems'))
            );
        },
        builder: (context, popup) {
            return Container(
                height: 150,
                child: popup,
            );
        },
        itemExtent: 40,
        numberofRowsAtSection: (section) {
            return 10;
        },
        itemBuilder: (section, row) {
            return Text('$row',style: TextStyle(fontSize: 12),);
        }
    );
}
```

### Custom PickerView
![image](https://github.com/CodeGeekXu/flutter_picker_view/blob/master/custom.jpeg)

```dart

PickerController controller = PickerController(count: 2);

PickerView(
    itemExtent: 40,
    numberofRowsAtSection: (section) {
        return 10;
    },
    itemBuilder: (section, row) {
        if (0 == section) {
            return Padding(
                padding: EdgeInsets.only(left: 20),
                child: Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: <Widget>[
                        Text('$row',style: TextStyle(fontSize: 14),),
                        Icon(Icons.face)
                    ],
                ),
            );
        } else {
            return Padding(
                padding: EdgeInsets.only(right: 20),
                child: Row(
                    mainAxisAlignment: MainAxisAlignment.center,
                    children: <Widget>[
                        Text('$row',style: TextStyle(fontSize: 14),),
                        Icon(Icons.thumb_up)
                    ],
                )
            );
        }
    },
    controller: controller
)
```



CHANGELOG.md
## [1.1.0] - 2020-0313
* support web platform

## [1.0.0] - 2019-09-26
* dart sdk support: ">=2.1.0 <3.0.0", remove Spread elements.

## [0.0.2] - 2019-09-17
* 解决PickerController selectedItems为null时PickerVIew不显示的bug

## [0.0.1] - 2019-0912
* A Flutter picker view like iOS UIPickerView support multiple pickers

example/lib/widgets/example_page.dart
import 'package:flutter/material.dart';
import 'package:flutter_picker_view/flutter_picker_view.dart';
import 'custom_picker_view.dart';

class ExamplePage extends StatefulWidget {
  
  @override
  State<StatefulWidget> createState() {
    return ExamplePageState();
  }
}

class ExamplePageState extends State<ExamplePage> {
  
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('flutter picker view'),),
      body: Container(
        alignment: Alignment.center,
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            _Button(
              title: 'AlertDialogPicker',
              onPressed: () {
                _showDialogPicker();
              }
            ),
            _Button(
              title: 'BottomSheetPicker',
              onPressed: () {
                _showBottomSheetPicekr();
              }
            ),
            _Button(
              title: 'CustomPicker',
              onPressed: () {
                _showCustomPicker();
              }
            )
          ],
        ),
      ),
    );
  }
  
  Widget _Button({String title, VoidCallback onPressed}) {
    return Container(
      margin: EdgeInsets.only(bottom: 50),
      child: FlatButton(
        color: Colors.blue,
        onPressed: onPressed,
        child: Text(title, style: TextStyle(color: Colors.white),)
      ),
    );
  }
  
  void _showDialogPicker() {
    
    PickerController pickerController = PickerController(count: 3, selectedItems: [5,2,1]);
    
    PickerViewPopup.showMode(
      PickerShowMode.AlertDialog,
      controller: pickerController,
      context: context,
      title: Text('alert dialog',style: TextStyle(fontSize: 14),),
      cancel: Text('cancel', style: TextStyle(color: Colors.grey),),
      onCancel: () {
        Scaffold.of(context).showSnackBar(
          SnackBar(content: Text('AlertDialogPicker.cancel'))
        );
      },
      confirm: Text('confirm', style: TextStyle(color: Colors.blue),),
      onConfirm: (controller) {
        List<int> selectedItems = [];
        selectedItems.add(controller.selectedRowAt(section: 0));
        selectedItems.add(controller.selectedRowAt(section: 1));
        selectedItems.add(controller.selectedRowAt(section: 2));
    
        Scaffold.of(context).showSnackBar(
          SnackBar(content: Text('AlertDialogPicker.selected:$selectedItems'))
        );
      },
      builder: (context, popup) {
        return Container(
          height: 280,
          child: popup,
        );
      },
      itemExtent: 40,
      numberofRowsAtSection: (section) {
        return 10;
      },
      itemBuilder: (section, row) {
        return Text('$row',style: TextStyle(fontSize: 12),);
      }
    );
  }
  
  void _showBottomSheetPicekr() {
    PickerController pickerController = PickerController(count: 3, selectedItems: [5,2,1]);
  
    PickerViewPopup.showMode(
      PickerShowMode.BottomSheet,
      controller: pickerController,
      context: context,
      title: Text('bottom sheet',style: TextStyle(fontSize: 14),),
      cancel: Text('cancel', style: TextStyle(color: Colors.grey),),
      onCancel: () {
        Scaffold.of(context).showSnackBar(
          SnackBar(content: Text('BottomSheetPicker.cancel'))
        );
      },
      confirm: Text('confirm', style: TextStyle(color: Colors.blue),),
      onConfirm: (controller) {
        List<int> selectedItems = [];
        selectedItems.add(controller.selectedRowAt(section: 0));
        selectedItems.add(controller.selectedRowAt(section: 1));
        selectedItems.add(controller.selectedRowAt(section: 2));
        
        Scaffold.of(context).showSnackBar(
          SnackBar(content: Text('BottomSheetPicker.selected:$selectedItems'))
        );
      },
      builder: (context, popup) {
        return Container(
          height: 250,
          child: popup,
        );
      },
      itemExtent: 40,
      numberofRowsAtSection: (section) {
        return 10;
      },
      itemBuilder: (section, row) {
        return Text('$row',style: TextStyle(fontSize: 12),);
      }
    );
  }
  
  void _showCustomPicker() {
    Navigator.of(context).push(MaterialPageRoute(builder: (context)=> CustomPickerView()));
  }
}